x-pack/metricbeat/Jenkinsfile.yml

```diff
@@ -20,6 +21,10 @@ stages:
           make check-no-changes;
     unitTest:
         mage: "mage build unitTest"
+        when:                  ## Override the top-level when.
+            not_changeset_full_match: "^metricbeat/.*" ## Disable the stage if ONLY changes for the OSS beats
```

Comment:
Same here (and with the other beats), will this prevent x-pack metricbeat tests to be run if there were changes in oss metricbeat only?

Reply:
Exactly, for clarity for the below `x-pack/metricbeat` CI stages:
- `lint` stage (L18) will always run as long as the first `when` condition is set (see L1)
- `unitTest` stage (L22) will run if the `changeset` does not contain ONLY `metricbeat` changes or it's a build in any branch/tag

`not_changeset_full_match` is a negative condition, exactly the opposite of what the majority of the other `when conditions`.

Reply:
But we want x-pack tests to be run if there are changes in some dependency in OSS (as detected by `GetProjectDependencies`).

e.g. if we change only something in `metricbeat/mb`, we want to run x-pack metricbeat tests as it can be affected.

Reply:
I'm gonna double check the expected behaviour works as expected, since I'm not convinced now if this might cause that particular regression :/

Reply:
Since there is no an easy way with the proposal, I finally change a bit the implementation to exclude the `x-pack/<beat>` stages if there are only `<beat>/docs` changes.

Reply:
> e.g. if we change only something in metricbeat/mb, we want to run x-pack metricbeat tests as it can be affected.

It should we now be covered. I just executed some tests for the `not_changeset_full_match` when condition locally:

```diff
diff --git a/src/test/groovy/BeatsWhenStepTests.groovy b/src/test/groovy/BeatsWhenStepTests.groovy
index 9f53c52..792a3a2 100644
--- a/src/test/groovy/BeatsWhenStepTests.groovy
+++ b/src/test/groovy/BeatsWhenStepTests.groovy
+  @Test
+  void test_xpackmetricbeat_test_when_metricbeat_mb_changes() throws Exception {
+    def script = loadScript(scriptName)
+    def changeset = "metricbeat/mb/my-file.go"
+    helper.registerAllowedMethod('readFile', [String.class], { return changeset })
+    def ret = script.whenNotChangesetFullMatch(content: [ not_changeset_full_match: '^metricbeat/docs/.*'])
+    printCallStack()
+    assertTrue(ret)
+  }
+
```

```bash
$ ./mvnw test -Dtest=BeatsWhenStepTests#test_xpackmetricbeat_test_when_metricbeat_mb_changes
...[INFO] Running BeatsWhenStepTests
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.981 s - in BeatsWhenStepTests
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```